osi_environment.proto

```diff
@@ -324,4 +336,150 @@ message EnvironmentalConditions
+            FRACTIONAL_CLOUD_COVER_FIVE_OKTAS = 7;
+
+            // 6/8 of the sky is covered with clouds.
+            //
+            FRACTIONAL_CLOUD_COVER_SIX_OKTAS = 8;
+
+            // 7/8 of the sky is covered with clouds.
+            //
+            FRACTIONAL_CLOUD_COVER_SEVEN_OKTAS = 9;
+
+            // 8/8 of the sky is covered with clouds.
+            //
+            FRACTIONAL_CLOUD_COVER_EIGHT_OKTAS = 10;
+
+            // Sky obscured, describes situations where the sky is not perceivable, e.g. in dense fog.
+            //
+            FRACTIONAL_CLOUD_COVER_SKY_OBSCURED = 11;
+        }
+    }
+
+    //
+    // \brief Defines wind properties.
+    //
+    message Wind
+    {
+        // The origin direction of the wind (not the target direction) in the ground/xy-plane of the
+        // world coordinate system. Corresponds to the heading/yaw angle, counted counterclockwise.
+        // 0 pointing north. If north is not explicitly defined via \c #osi3::GroundTruth::proj_string,
+        // then north is pointing in y-axis direction. The default orientation (x, y, z) is easting, northing, up [1].
+        //
+        // \note 0: north; +pi/2: west; pi: south, 3/2 pi: east.
+        //
+        // Unit: rad
+        //
+        // The preferred angular range is [0, 2pi].
+        //
+        // \par References:
+        // [1] PROJ contributors. (2019). <em>PROJ coordinate transformation software library</em>. Open Source Geospatial Foundation. Retrieved January 25, 2019, from https://proj.org/usage/projections.html
+        //
+        optional double origin_direction = 1;
+
+        // The wind speed.
+        //
+        // Unit: m/s
+        //
+        // \rules
+        // is_greater_than_or_equal_to: 0
+        // \endrules
+        //
+        optional double speed = 2;
+    }
+
+    //
+    // \brief Properties of the sun.
+    //
+    message Sun
+    {
+        // Azimuth of the sun, counted counterclockwise.
+        // 0 pointing north. If north is not explicitly defined via \c #osi3::GroundTruth::proj_string,
+        // then north is pointing in y-axis direction. The default orientation (x, y, z) is easting, northing, up [1].
+        //
+        // \note 0: north; +pi/2: west; pi: south, 3/2 pi: east.
+        //
+        // Unit: rad
+        //
+        // The preferred angular range is [0, 2pi].
+        //
+        // \par References:
+        // [1] PROJ contributors. (2019). <em>PROJ coordinate transformation software library</em>. Open Source Geospatial Foundation. Retrieved January 25, 2019, from https://proj.org/usage/projections.html
+        //
+        optional double azimuth = 1;
+
+        // Solar elevation angle. Counted in positive direction (clockwise) around xy-plane.
+        //
+        // \note 0: xy-plane; +pi/2: zenith.
+        //
+        // Unit: rad
+        //
+        // The preferred angular range is [-pi/2, +pi/2].
+        //
+        optional double elevation = 2;
```

Comment:
Regarding Azimuth and Elevation. We are using functions that calculate this boths values as a result of daytime and date:
f(daytime,date)=azimuth
f(daytime,date)=elevation
So it may be redundant here - also I know not everyone is using this mechanism.
From our perspective, we don't need azimuth and elevation, but so far we miss the date.

Reply:
Addition: Having the date, it would also help us to determine the season (spring, summer, autumn, winter).

Reply:
Time of day and unix timestamp do already exist in EnvironmentalConditions (if those are what you need).

Reply:
Result from harmonization meeting (08.04.22): it is indeed redundant information if proj4 string and unix timestamp is set correctly. But not in all use cases the data is available, e.g. for artificial roads without correct location. Even if everything is available, it could be hard to calculate the sun position based on localization and unix time stamp for external sensor models. So we would propose to keep this information here.

Reply:
The redundancy (in our use case) is acceptable, so that doesn't hurt.
The problem left is the date:
Does the unix timestamp represent the date the simulation takes place? 
If so, a better description would be good.
If not (and what I thought), is it the timestamp of message?
For example: We want to have Winter, so we would send a date like 15th of Dec. In our graphic engine everywhere is snow. But of course in reality today we have 11th of April. So is the timestamp free choosable?

Reply:
(And the sun is changing its elevation accordingly)

Reply:
I think, since the field unix_timestamp is located in EnvironmentalConditions it should be set to the current UTC date (+time?) at the host vehicle's current location. The description of EnvironmentalCondition specifically note **"These conditions apply locally around the host vehicle."**. So I would assume that this applies to all included fields.

I did some research on when and how this field was introduced to OSI: Initially, there was a field called Date (see early commits in https://github.com/OpenSimulationInterface/open-simulation-interface/pull/323 e.g., https://github.com/OpenSimulationInterface/open-simulation-interface/pull/323/commits/aedfaec5d39db3e44cdb45d61a9aa470b937671b) with the comment **"The date at the host vehicles location."** In https://github.com/OpenSimulationInterface/open-simulation-interface/pull/323 this was then changed into the field unix_timestamp and the sentence somehow disappeared.

But probably we should discuss if it really means what I interpreted and add some details to the field's comment in any case.